glean-core/csharp/Glean/Glean.cs

```diff
@@ -104,13 +110,15 @@ string dataDir
                     return;
                 }
 
-                /* TODO:
-                // If any pings were registered before initializing, do so now.
-                // We're not clearing this queue in case Glean is reset by tests.
-                synchronized(this@GleanInternalAPI) {
-                    pingTypeQueue.forEach { registerPingType(it) }
+                //// If any pings were registered before initializing, do so now.
+                //// We're not clearing this queue in case Glean is reset by tests.
+                lock (pingTypeQueue)
```

Comment:
```suggestion
                lock (this)
```

This should lock using the same object as the `[MethodImpl(MethodImplOptions.Synchronized)]` in the methods, otherwise we'll have races.